src/Analyzer/Identifier.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Analyzer/Identifier.h>
+#include <base/defines.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+
+namespace DB
+{
+
+/// Create Identifier from parts
+Identifier::Identifier(const std::vector<std::string> & parts_) : parts(parts_), full_name(boost::algorithm::join(parts, "."))
+{
+}
+
+/// Create Identifier from parts
+Identifier::Identifier(std::vector<std::string> && parts_) : parts(std::move(parts_)), full_name(boost::algorithm::join(parts, "."))
```

Comment:
```suggestion
Identifier::Identifier(std::vector<std::string> && parts_)
    : parts(std::move(parts_))
    , full_name(boost::algorithm::join(parts, "."))
```

Reply:
Clang format changes it to this style. I'll check what needs to be changed in its config to produce the format you suggest instead